src/arch/x86/isa/microops/fpop.isa

```diff
@@ -257,7 +257,14 @@ let {{
             super().__init__(reg1, reg2, reg3, **kwargs)
 
     class Movfp(Fp2Op):
-        code = 'FpDestReg_uqw = FpSrcReg1_uqw;'
+        code = '''
+        if(dataSize == 4) {
+            FpDestReg_uqw = (0xFFFFFFFF00000000 & FpDestReg_uqw) |
+            (0x00000000FFFFFFFF & FpSrcReg1_uqw);
```

Comment:
Can you use something from `bitfield.hh`? E.g., `bits(FpDestReg_uqw, 32, 63);`

Reply:
Gotcha, thanks! I'll push up a fix that uses the bits() call.